### DISCUSSION TITLE
Difference between CONNECTED_SMS and CONNECTED_DATA

### DISCUSSION THREAD

**Discussion by StefanoFalsetto-CKHIOD** [camaraproject/DeviceStatus#135] [Q&A] [answered]
I need help understanding which kind of data must be used to return the two values in subject.
The YAML file is stating:

```
      - `CONNECTED_SMS`, if device is connected to the network via SMS usage
      - `CONNECTED_DATA`, if device is connected to the network via data usage

```

Can someone please elaborate a bit more the description?
Is CONNECTED_DATA a status obtained from  HLR query MAP-SEND-ROUTING-INFO and CONNECTED_SMS obtained from HLR query MAP-SEND-ROUTING-INFO-FORM-SM?

Thank you.

**Comment by akoshunyadi** [comment]
Hi, the implementation is based on the 3gpp MonitoringEvent API, with following definition:
        - SMS : The SCS/AS requests to be notified when the UE becomes reachable for sending SMS to the UE
        - DATA: The SCS/AS requests to be notified when the UE becomes reachable for sending downlink data to the UE

**Comment by StefanoFalsetto-CKHIOD** [comment]
Thank you @akoshunyadi for your reply.

What if an MNO does not export NEF functionalities? Is this service not compatible with this kind of MNOs?

**Reply by nunopires2001** [reply]
I know this is a old thread, but I will reply anyway since I have been discussing this subject internally.
When a NEF is not in place, the solution should be to query directly network nodes.
A possible solution is to query Central Databases, like UDM and PCRF.
CONNECTED_DATA: It´s possible to query PCRF to receive this information, since PCRF tracks subscriber data sessions. 
CONNECTED_SMS: It's possible to query UDM about subscriber connection to network and if it has SMS service provisioned.

**Reply by StefanoFalsetto-CKHIOD** [reply]
Thank you @nunopires2001 for the answer, and I am very sorry for my late answer. I didn't received any notification from github about this comment...

So, in case that an MNO is not willing to invest in exporting PCRF and UDM data or NEF functionalities, is it safe to assume that this service cannot be implemented? Am I right?
I mean, the information I need to pass to MNOs is that this service cannot be implemented just using the HLR to retrieve the connectivity status but it is mandatory to use of NEF or UDM/PCRF query.

**Reply by nunopires2001** [reply, marked as answer]
Hi Stefano,
If your client/partner is limited to HLR exposure they best you can achieve is to provide CONNECTED_SMS information.
You need to have access to packet core functions in order to have information regarding data mobile sessions. In 4G and 5G, PCRF could be a could option if charging policies are used for whole data subscribers. If not  you should need to query a central database, like UDM + HSS/CUDB.

Anyway these are not so standard ways to exposure the network resources. The standard way is to use SCEF (4G) and NEF (5G). These functions implement the necessary integrations to network platforms and expose the info on REST APIs ready to be exposed on a Camara layer.

**Reply by StefanoFalsetto-CKHIOD** [reply]
Thank you @nunopires2001 for this clarification, that's quite useful to me.
